Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -647,6 +647,32 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     }
     return .visitChildren
   }
+  
+  public override func visit(_ node: IfConfigDeclSyntax) -> SyntaxVisitorContinueKind {
+    for clause in node.clauses where clause.hasError {
+      if let unexpectedBeforePoundKeyword = clause.unexpectedBeforePoundKeyword,
+         clause.poundKeyword.tokenKind == .poundElseifKeyword,
+         clause.poundKeyword.presence == .missing {
+        let unexpectedTokens = unexpectedBeforePoundKeyword
+          .suffix(2)
+          .compactMap { $0.as(TokenSyntax.self) }
+        if unexpectedTokens.map(\.tokenKind) == [.poundElseKeyword, .keyword(.if)] {
```

Comment:
I think the two `if`s can again be merged

```suggestion
          .compactMap { $0.as(TokenSyntax.self) },
          unexpectedTokens.map(\.tokenKind) == [.poundElseKeyword, .keyword(.if)] {
```

Reply:
The problem I see here is that the `unexpectedTokens` constant isn't part of the if statement above, and it's not `Optional`, so I can't include all of them in the same if statement:
```swift
  if let unexpectedBeforePoundKeyword = clause.unexpectedBeforePoundKeyword,
     clause.poundKeyword.tokenKind == .poundElseifKeyword,
     clause.poundKeyword.presence == .missing { // End of first if statement
       let unexpectedTokens = unexpectedBeforePoundKeyword // Constant declaration (non-optional)
         .suffix(2)
         .compactMap { $0.as(TokenSyntax.self) }
       if unexpectedTokens.map(\.tokenKind) == [.poundElseKeyword, .keyword(.if)] { // Second if statement
         // ...
       }
```

Reply:
Oh, sorry, I misread that on GitHub. You’re absolutely right. We can’t merge the `if`s in that case.